doc/api.md

```diff
@@ -0,0 +1,422 @@
+
+<!--
+TODO:
+Alphabetic sort for each function?
+-->
```

Comment:
You mean whether the functions should be listed alphabetically inside each section? No I don't think that's necessary. I'd rather have them grouped logically in whatever order makes most sense.